src/topology/vector_bundle.lean

```diff
@@ -249,3 +249,138 @@ lemma is_topological_vector_bundle_is_topological_fiber_bundle :
 λ x, ⟨(trivialization_at R F E x).to_bundle_trivialization, mem_base_set_trivialization_at R F E x⟩
 
 end topological_vector_bundle
+
+/-! ### Constructing topological vector bundles -/
+
+variables (B)
+
+/-- Analogous construction of `topological_fiber_bundle_core` for vector bundles. -/
+@[nolint has_inhabited_instance]
+structure topological_vector_bundle_core (ι : Type*) :=
+(base_set          : ι → set B)
+(is_open_base_set  : ∀i, is_open (base_set i))
+(index_at          : B → ι)
+(mem_base_set_at   : ∀x, x ∈ base_set (index_at x))
+(coord_change  : ι → ι → B → (F →ₗ[R] F))
+(coord_change_self : ∀i, ∀ x ∈ base_set i, ∀v, coord_change i i x v = v)
+(coord_change_continuous : ∀i j, continuous_on (λp : B × F, coord_change i j p.1 p.2)
+                                               (set.prod ((base_set i) ∩ (base_set j)) univ))
+(coord_change_comp : ∀i j k, ∀x ∈ (base_set i) ∩ (base_set j) ∩ (base_set k), ∀v,
+  (coord_change j k x) (coord_change i j x v) = coord_change i k x v)
+
+attribute [simp, mfld_simps] topological_vector_bundle_core.mem_base_set_at
+
+namespace topological_vector_bundle_core
+
+variables {R B F} {ι : Type*} (Z : topological_vector_bundle_core R B F ι)
+
+/-- Natural identification to a `topological_fiber_bundle_core`. -/
+def to_topological_vector_bundle_core : topological_fiber_bundle_core ι B F :=
+{ coord_change := λ i j b, Z.coord_change i j b, ..Z }
+
+instance to_topological_vector_bundle_core_coe : has_coe (topological_vector_bundle_core R B F ι)
+  (topological_fiber_bundle_core ι B F) := ⟨to_topological_vector_bundle_core⟩
+
+include Z
+
+lemma coord_changeear_comp (i j k : ι): ∀ x ∈ (Z.base_set i) ∩ (Z.base_set j) ∩ (Z.base_set k),
+  (Z.coord_change j k x).comp (Z.coord_change i j x) = Z.coord_change i k x :=
+λ x hx, by { ext v, exact Z.coord_change_comp i j k x hx v }
+
+/-- The index set of a topological vector bundle core, as a convenience function for dot notation -/
+@[nolint unused_arguments has_inhabited_instance]
+def index := ι
+
+/-- The base space of a topological vector bundle core, as a convenience function for dot notation-/
+@[nolint unused_arguments, reducible]
+def base := B
+
+/-- The fiber of a topological vector bundle core, as a convenience function for dot notation and
+typeclass inference -/
+@[nolint unused_arguments has_inhabited_instance]
+def fiber (x : B) := F
+
+instance topological_space_fiber (x : B) : topological_space (Z.fiber x) :=
+by { dsimp [fiber], apply_instance }
```

Comment:
instances using `dsimp` are bad, because they introduce an `id` in the term that can prevent further unification later on. (You can see it if you `#print topological_vector_bundle_core.topological_space_fiber`: the last line of the output is `id _inst_4`).

To avoid this, work in a section and set `local attribute [reducible] fiber`.

Reply:
I am not sure I understand this comment: should I take out this instance completely then? Doesn't the same apply to the same instance in `topological_fiber_bundle`?

Reply:
Why can't I then write
```
instance topological_space_fiber (x : B) : topological_space (Z.fiber x) := _inst_4
```
or something of this kind?

Reply:
Writing `_inst_4` is bad because it will break if someone changes the order of variables, for instance. You should do what I said, i.e., work in a section and prove your instance there. I mean,
```lean

section
local attribute [reducible] fiber

instance topological_space_fiber (x : B) : topological_space (Z.fiber x) :=by apply_instance

end
```
And indeed the ones in `topological_fiber_bundle` should also be fixed in the same way.

Reply:
Oh now I see what you mean! I thought you meant that anytime one needed to use this construction should have worked in a section by declaring `local attribute [reducible] fiber` and I was like "it's super weird Sébastien wants to do this!"

Reply:
I am surprised this works but I see now you adopted this trick also for tangent bundle now whereas you used to name instances last year